kafka/src/test/java/nl/nn/adapterframework/extensions/kafka/KafkaSenderTest.java

```diff
@@ -0,0 +1,84 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.extensions.kafka;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.apache.kafka.clients.producer.MockProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import lombok.SneakyThrows;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.PipeLineSession;
+import nl.nn.adapterframework.stream.Message;
+
+public class KafkaSenderTest {
+	final MockProducer<String, byte[]> mockProducer = new MockProducer<>(true, new StringSerializer(), new ByteArraySerializer());
+	final KafkaSender sender = new KafkaSender();
+	@BeforeEach
+	@SneakyThrows
+	void setUp() {
+		sender.setTopic("test.test2");
+		sender.setClientId("test");
+		sender.setBootstrapServers("example.com:9092"); //dummy, doesn't connect.
+		sender.configure();
+		sender.getInternalSender().setProducer(mockProducer);
+	}
+
+	@Test
+	@SneakyThrows
+	public void test() {
+		Message message = new Message("Hello World");
+		PipeLineSession session = new PipeLineSession();
+		sender.sendMessage(message, session);
+		assertEquals(mockProducer.history().size(), 1, "One message should be sent");
+		ProducerRecord<String, byte[]> test = mockProducer.history().get(0);
+		assertEquals(test.topic(), "test.test2", "Topic should be test.test2");
+		assertArrayEquals(test.value(), "Hello World".getBytes(), "Message should be set");
+	}
+
+	@ParameterizedTest
+	@MethodSource
+	void validateParameters(Consumer<KafkaFacade> consumer, boolean shouldSucceed, String name) {
+		consumer.accept(sender);
+		if(shouldSucceed) Assertions.assertDoesNotThrow(sender::configure, name);
+		else Assertions.assertThrows(ConfigurationException.class, sender::configure, name);
+	}
+	public static Consumer<KafkaSender> wrap(Consumer<KafkaSender> function) {
+		return function;
+	}
+	static Stream<Arguments> validateParameters() {
+		return Stream.of(
+				Arguments.of(wrap(sender->sender.setTopic(null)), false, "null topic"),
+				Arguments.of(wrap(sender->sender.setTopic("")), false, "empty topic"),
+				Arguments.of(wrap(sender->sender.setTopic("test")), true, "valid topic"),
+				Arguments.of(wrap(sender->sender.setTopic("test,test2")), false, "multiple topics"),
+				Arguments.of(wrap(sender->sender.setTopic("test*")), false, "wildcard topic")
```

Comment:
Waarom is `sender.setTopic(...)` deel van alle arguments here? Met een `Consumer<T>` argument wat makkelijk wordt verward met Kafka Consumer?

Waarom maak je deze logic niet gewoon deel van de test, en de topic-string een parameter?
Dan kan deze `MethodSource` ook vervangen door een `CsvSource` die meteen boven de method staat en daarom het geheel overzichtelijker maakt.